apps/src/lib/util/StatsigReporter.js

```diff
@@ -12,6 +12,17 @@ const NO_EVENT_NAME = 'NO_VALID_EVENT_NAME_LOG_ERROR';
 
 class StatsigReporter {
   constructor() {
+    let user = {};
+    user_id_element = document.querySelector('script[data-user-id]');
```

Comment:
We'll want to be careful about relying on this on cached pages, where it'll become out of date. The nice thing about using userRedux is that it's updated for every page load so it was always (eventually) up to date. 

I'm curious if we can still use the user passed through from userRedux, but save that user id in this class so that we can send it with events? Or is there a race condition there that would make that difficult?

Reply:
Can we pair on this? It seems that the current user redux has been updating Amplitude on init(), which doesn't seem to get called for each page load.

I'm also surprised that a page would be cached through a change in user log out/sign in. This would make me think that some users could access data from a previously logged in user's cache. Is that the case or am I misunderstanding?

Reply:
oh weird -- it's probably not great if it doesn't get called on each page load 😬 

On cached pages (and also I think on uncached pages as a result), we fetch the user data asyncronously, which then is fed into the redux store [here](https://github.com/code-dot-org/code-dot-org/blob/159aa5956c4a92afa0df420228453f60c7bb4109/apps/src/code-studio/header.js#L219). Most of our pages aren't cached, so in most cases, this solution is fine, but it's always possible we accidentally try to use this on one of those cached page and become confused.